src/track/beat.cpp

```diff
@@ -0,0 +1,48 @@
+#include "beat.h"
+
+namespace mixxx {
+Beat::Beat(FramePos framePos,
+        BeatType type,
+        const TimeSignature& timeSignature,
+        const Bpm& bpm,
+        int beatIndex,
+        int barIndex,
+        uint beatInBarIndex,
+        BeatMarkers markers)
+        : m_framePos(framePos),
+          m_eType(type),
```

Comment:
Should this be a QFlags field instead? Maybe there are other kinds of beat highlights, e. g. Phase/section start.

Reply:
With this PR, I'll let it be. I'll change this to `QFlags` while implementing sections and phrases.